### PR TITLE
Restructure and document XML generation functions. 

### PR DESCRIPTION
**Pull request by JPTIZ**

### CODE REVIEW

File: src/main/webapp/js/grapheditor/Graph.js [hunk 1]
```diff
@@ -28,8 +28,10 @@ function parseStyle(style) {
   const shape = jsonStyle.shape;
   if (shape) {
     const customProperties = mxCellRenderer.defaultShapes[shape].prototype.customProperties;
-    for (const {name, defVal} of customProperties) {
-      jsonStyle[name] = jsonStyle[name] || defVal;
+    if (customProperties != null) {
+      for (const {name, defVal} of customProperties) {
+        jsonStyle[name] = jsonStyle[name] || defVal;
```
Comment (Coquinho) [inline src/main/webapp/js/grapheditor/Graph.js:33]: I think that way we cant set a property to `false` if its `defVal` is `true, maybe something like:
```suggestion
        if (typeof(jsonStyle[name]) === 'undefined') { 
          jsonStyle[name] = defVal;
        }
```

**Review by Coquinho — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/grapheditor/Graph.js (hunk 1 above)
Comment (JPTIZ) [reply src/main/webapp/js/grapheditor/Graph.js:33]: Fixed :)

Comment (Coquinho) [reply src/main/webapp/js/grapheditor/Graph.js:33]: sorry, but don't think so. We still have the problem of setting a property to `true` if its `defVal` is `true` while we are setting it to `false`.

Comment (JPTIZ) [reply src/main/webapp/js/grapheditor/Graph.js]: I changed the wrong part of the code :P I'll wait for your thumbs up before setting as resolved this time.

**Pull request merged**